java-checks/src/test/files/checks/security/HostnameVerifierImplementationCheck.java

```diff
@@ -0,0 +1,72 @@
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLSession;
+
+class A {
+  private B b;
+
+  A(B b) {
+    this.b = b;
+  }
+
+  void method1() {
+    return b.method(new HostnameVerifier() {
+      @Override
+      public boolean verify(String a, SSLSession b) {
+        return true; // Noncompliant {{Do not unconditionally return true in this method.}}
+      }
+    });
+  }
+
+  void method2() {
+    return b.method((a, b) -> true); // Noncompliant
```

Comment:
Your implementation won't catch a case where lambda is assigned to a variable. You should not check for lambda as argument but for every lambda of the proper type (hostnameVerifier) check if it is compliant or not.